src/base/search/searchhandler.cpp

```diff
@@ -127,21 +127,25 @@ void SearchHandler::processFinished(const int exitcode)
 void SearchHandler::readSearchOutput()
 {
     QByteArray output = m_searchProcess->readAllStandardOutput();
-    output.replace("\r", "");
+    output.replace('\r', "");
+
     QList<QByteArray> lines = output.split('\n');
     if (!m_searchResultLineTruncated.isEmpty())
         lines.prepend(m_searchResultLineTruncated + lines.takeFirst());
     m_searchResultLineTruncated = lines.takeLast().trimmed();
 
-    QList<SearchResult> searchResultList;
+    QVector<SearchResult> searchResultList;
+    searchResultList.reserve(lines.size());
+
     for (const QByteArray &line : asConst(lines)) {
         SearchResult searchResult;
         if (parseSearchResult(QString::fromUtf8(line), searchResult))
             searchResultList << searchResult;
     }
 
     if (!searchResultList.isEmpty()) {
-        m_results.append(searchResultList);
+        for (const SearchResult &result : searchResultList)
```

Comment:
Last question. Would it be overkill if you added `m_results.reserve(m_result.size() + searchResultList.size())` before?

Reply:
I think yes, I printed out the searchResultList size, it jumps between 1~36. Reserving small ranges each time isn't much different than just appending to it.

Reply:
Ok, let's finish it!